pkg/modelfinder/modelfinder.go

```diff
@@ -0,0 +1,169 @@
+package modelfinder
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+var prefixListToIgnore = []string{".cog", ".git", "__pycache__"}
+
+var suffixListToIgnore = []string{
+	".py", ".ipynb", ".whl", // python projects
+	".jpg", ".jpeg", ".png", ".webp", ".svg", ".gif", ".mp4", // images and videos
+	".log", // logs
+}
+
+// ModelFinder is an interface that finds all the model files in the current directory.
+type ModelFinder interface {
+	// FindModels finds all the model files in the current directory.
+	// It returns a list of directories (except current directory) that contain model files,
+	// and a list of model files in current directory.
+	FindModels() ([]string, []string, error)
+}
+
+type FileWalker func(root string, walkFn filepath.WalkFunc) error
+
+type modelFinder struct {
+	fw FileWalker
+}
+
+func New() ModelFinder {
+	return &modelFinder{
+		fw: filepath.Walk,
+	}
+}
+
+func (mf *modelFinder) FindModels() ([]string, []string, error) {
```

Comment:
This method is called `FindModels`, but the scratch image we write to is called `modelweights`. Is there a single term that accurately describes what we're looking for?